py-tokenizers: add v0.22.1

qwertos opened the pull request.

qwertos committed:
[py-tokenizers] added version 0.22.1

tldahlgren commented on repos/spack_repo/builtin/packages/py_tokenizers/package.py:
```diff
@@ -24,7 +25,8 @@ class PyTokenizers(PythonPackage):
 
     # TODO: This package currently requires internet access to install.
```
FYI. The archive from `PyPi.org` indicates the minimum python for the new version is `python@3.9:`.

tldahlgren commented on repos/spack_repo/builtin/packages/py_tokenizers/package.py:
```diff
@@ -24,7 +25,8 @@ class PyTokenizers(PythonPackage):
 
     # TODO: This package currently requires internet access to install.
     depends_on("py-maturin@1", when="@0.14:", type="build")
```
FWIW. Based on prior comments from Adam, I'm ignoring the upper bound for `py-maturin` in the `pyproject.toml` file.

tldahlgren reviewed (changes requested):
Confirmed the version sha256. 

Note the `pyproject.toml` downloaded from PyPi has a lower bound for `python` still listed as a safe version. Since someone may already have a version of `python` installed, I would think it would be prudent to include the lower bound.

qwertos committed:
[py-tokenizers]

per https://github.com/spack/spack-packages/pull/1860#discussion_r2408990337 removing upper bound on py-maturin

qwertos committed:
[py-tokenizers] added python minimum bound for version 0.22.1
